File: P2/java/deepspace/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

class ShieldBooster{
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String n, float b, int u) {
        name = n;
        boost = b;
        uses = u;
    }
    
    ShieldBooster(final ShieldBooster other) {
        name = other.name;
        boost = other.boost;
        uses = other.uses;
    }
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
    
    public String toString(){   //NO SE SI ESTA BIEN
        return ("name: " + name.toString() + ", boost: " + boost + ", uses: " + uses);
    }
    
    float getBoost(){
        return boost;
    }
    
    int getUses(){
        return uses;
    }
    
    public float useIt() {
        if(uses > 0){
            uses -= 1;
            return boost;
        }else
            return 1;
    }
}

File: P3/ruby/ShieldBooster.rb
# encoding: UTF-8

require_relative 'ShieldToUI'

module Deepspace
	class ShieldBooster
		def initialize (n, b, u) # ShieldBooster(n : string, b : float, u : int)
			@name = n
			@boost = b
			@uses = u
		end
		
		attr_reader :name
		attr_reader :boost
		attr_reader :uses
		
		def self.newCopy (other)
			new(other.name, other.boost, other.uses)
		end
		
		def getUIversion
			ShieldToUI.new(self)
		end

		def to_s
			getUIversion.to_s 
		end
		
		def useIt
			if @uses > 0
				@uses -= 1
				@boost
			else
				1.0
			end
		end
	end
end

if $0 == __FILE__
	puts Deepspace::ShieldBooster.new(1, 2, 3)
end
